### PR TITLE
Create code-actions to generate interceptor method template

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

Introduces two code-actions to generate interceptor method template : 
1. `Add interceptor resource method` - for `RequestInterceptor` and `RequestErrorInterceptor`
2. `Add interceptor remote method` - for `ResponseInterceptor` and `ResponseErrorInterceptor`

The following templates will be added to interceptor services through these code-actions :
1. `RequestInterceptor` : 
```ballerina
resource function 'default [string... path](http:RequestContext ctx) 
       returns http:NextService|error? {
    // Add your logic here
    return ctx.next();
}
```
2. `RequestErrorInterceptor` : 
```ballerina
resource function 'default [string... path](error err, http:RequestContext ctx) 
       returns http:NextService|error? {
    // Add your logic here
    return ctx.next();
}
```
3. `ResponseInterceptor` : 
```ballerina
remote function interceptResponse(http:RequestContext ctx) 
       returns http:NextService|error? {
    // Add your logic here
    return ctx.next();
}
```
4. `ResponseErrorInterceptor` : 
```ballerina
remote function interceptResponseError(error err, http:RequestContext ctx) 
       returns http:NextService|error? {
    // Add your logic here
    return ctx.next();
}
```

> Fixes [`Create code-action to generate interceptor method template #2664`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2664)

## Example

https://user-images.githubusercontent.com/63336800/165223394-591ce462-57d0-4a18-932c-cfa60242685f.mov

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] <s>Updated the spec</s>

### CODE REVIEW

**Comment by shafreenAnfar** [comment]
This is great!

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result1.bal [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+service class RequestInterceptor {
+	*http:RequestInterceptor;
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result1.bal:20]: indentation issue?

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_3/service.bal [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+service class RequestInterceptor {
+	*http:RequestInterceptor;
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_3/service.bal:20]: indentation issue?

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result2.bal [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import ballerina/http;
+
+service class RequestInterceptor {
+	*http:RequestInterceptor;
+}
+
+service class RequestErrorInterceptor {
+	*http:RequestErrorInterceptor;
+
+	resource function 'default [string... path](error err, http:RequestContext ctx) returns http:NextService|error? {
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result2.bal:26]: Indentation issue?

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result3.bal [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+service class RequestErrorInterceptor {
+	*http:RequestErrorInterceptor;
+}
+
+service class ResponseInterceptor {
+	*http:ResponseInterceptor;
+
+	remote function interceptResponse(http:RequestContext ctx) returns http:NextService|error? {
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result3.bal:30]: Indentation issue

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result4.bal [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+service class ResponseInterceptor {
+	*http:ResponseInterceptor;
+}
+
+service class ResponseErrorInterceptor {
+	*http:ResponseErrorInterceptor;
+
+	remote function interceptResponseError(error err, http:RequestContext ctx) returns http:NextService|error? {
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result4.bal:34]: Indentation issue

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result1.bal (hunk 1 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result1.bal:23]: Added a comment to the file in [36a803d](https://github.com/ballerina-platform/module-ballerina-http/pull/992/commits/36a803d3b71ae67a41c896c3358f071f9f13a24e)

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_3/service.bal (hunk 2 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_codeaction_package_3/service.bal:23]: Added a comment to the file in [36a803d](https://github.com/ballerina-platform/module-ballerina-http/pull/992/commits/36a803d3b71ae67a41c896c3358f071f9f13a24e)

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result2.bal (hunk 3 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result2.bal:29]: Added a comment to the file in [36a803d](https://github.com/ballerina-platform/module-ballerina-http/pull/992/commits/36a803d3b71ae67a41c896c3358f071f9f13a24e)

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result4.bal (hunk 5 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result4.bal:37]: Added a comment to the file in [36a803d](https://github.com/ballerina-platform/module-ballerina-http/pull/992/commits/36a803d3b71ae67a41c896c3358f071f9f13a24e)

File: compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result3.bal (hunk 4 above)
Comment (TharmiganK) [reply compiler-plugin-tests/src/test/resources/codeaction/add_interceptor_method/result3.bal:33]: Added a comment to the file in [36a803d](https://github.com/ballerina-platform/module-ballerina-http/pull/992/commits/36a803d3b71ae67a41c896c3358f071f9f13a24e)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**